### PR TITLE
Update SearchGUI 4.0.4 final release & PeptideShaker 2.0.5 final release

### PR DESCRIPTION
**Pull request by CarlosHorro**
- New final releases:
SearchGUI 4.0.4
PeptideShaker 2.0.5

- Most important features:
support for mzML files
Added ThermoRawFileParser for support for Thermo raw file conversion
Required JDK 11 or higher

- For a full description of all changes:

https://github.com/compomics/peptide-shaker-2.0-issue-tracker/wiki/ReleaseNotes
(MetaMorpheus support will be added in a later version to the Conda package)

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

For members of the Bioconda project, the following command is also available:

<table>
  <tr>
    <td><code>@BiocondaBot please merge</code></td>
    <td>Upload built packages/containers and merge a PR. <br />Someone must approve a PR first! <br />This reduces CI build time by reusing built artifacts.</td>
  </tr>
</table>

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

**Comment by CarlosHorro** [comment]
Hi @bioconda/core,

The testing environment failed solving some dependencies ("conda.exceptions.UnsatisfiableError: The following specifications were found to be incompatible with each other") for SearchGUI, which is surprising as I successfully built searchgui Conda package in my local. Could you help me, please?


Greetings,
Carlos

**Comment by dpryan79** [comment]
There's a dependency conflict between dotnet and mono somewhere. I've only tracked it down to a conflict between mono itself and dotnet-runtime. Most likely something in conda-forge needs to be rebuilt.

File: recipes/searchgui/meta.yaml [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
-{% set version = "4.0.1.alpha" %}
+{% set version = "4.0.0" %}
 
 package:
   name: searchgui
   version: {{ version }}
 
 build:
-  number: 0
+  number: 1
```
Comment (bgruening) [inline recipes/searchgui/meta.yaml:10]: reset to 0

File: recipes/searchgui/meta.yaml [hunk 2]
```diff
@@ -1,25 +1,27 @@
 # Do not forget to update the version string in the searchgui.py file
 {% set name = "SearchGUI" %}
-{% set version = "4.0.1.alpha" %}
+{% set version = "4.0.0" %}
```
Comment (bgruening) [inline recipes/searchgui/meta.yaml:3]: this will cause problems, as the latest version will still be the 4.0.1

File: recipes/searchgui/meta.yaml [hunk 3]
```diff
@@ -1,25 +1,27 @@
...
 
 requirements:
   build:
     - {{ compiler('c') }}
   run:
-    - openjdk >=8
+    - openjdk >=11
     - mono >=4.0.0
     - python
+    - dotnet
```
Comment (bgruening) [inline recipes/searchgui/meta.yaml:24]: is dotnet __and__ mono needed?

Comment (CarlosHorro) [reply recipes/searchgui/meta.yaml:24]: @bgruening Hi! Yes, I know it sounds weird but they are. dotnet is required to run new MetaMorpheus search engine, and mono to run Andromeda (which fails using dotnet).

File: recipes/searchgui/meta.yaml (hunk 2 above)
Comment (CarlosHorro) [reply recipes/searchgui/meta.yaml:3]: Um...ok, then I will update the app to last bugfix version 4.0.3...

**Comment by CarlosHorro** [comment]
> There's a dependency conflict between dotnet and mono somewhere. I've only tracked it down to a conflict between mono itself and dotnet-runtime. Most likely something in conda-forge needs to be rebuilt.

Thank you @dpryan79 and @bgruening  . What I don't understand is why I'm able to build searchgui conda package in my local without any problem (I already uploaded it to an own conda channel), but its dependencies resolution fail in this environment.

By the way, I tried to execute circleci in my local too to replicate as much as possible the entire process but I always get this:
```

bioconda-utils build --docker --mulled-test --git-range master

12:19:47 BIOCONDA INFO Considering total of 7540 recipes.
12:19:48 BIOCONDA WARNING Multiple remotes found. Using first
12:19:48 BIOCONDA INFO Constraining to 0 git modified recipes.
12:19:48 BIOCONDA INFO Processing 0 recipes.
12:19:48 BIOCONDA INFO Nothing to be done.
```

**Comment by dpryan79** [comment]
I only ever use the `--packages` option with bioconda-utils locally, since I've had random issues like that with `--git-range master
` locally (probably due to not having updated the master branch.

File: recipes/peptide-shaker/meta.yaml [hunk 4]
```diff
@@ -1,25 +1,26 @@
 # Do not forget to update the version string in the peptide-shaker.py file
 {% set name = "PeptideShaker" %}
-{% set version = "2.0.1.alpha" %}
+{% set version = "2.0.0" %}
```
Comment (bgruening) [inline recipes/peptide-shaker/meta.yaml:3]: you will have the same problem here.

**Comment by bgruening** [comment]
dotnet needed a more recent glibc version afaik and this is causing trouble in our "older" build env, afaik. @mbargull do you know more about this by any chance?

**Comment by mbargull** [comment]
We need to update our build containers to support this.
Hopefully we'll be able to upgrade our whole setup in the near-ish future (early next year).
But depending on how important/urgent this is, we can think about upgrading to CentOS 7 images even before we raise our compatibility requirement to CentOS 7 / glibc >=2.17.
The reason why/how that would be possible is that we currently link against glibc 2.12 (and not the system's one) in any case (through the `sysroot_linux-64` package we use via the compiler packages).

**Comment by CarlosHorro** [comment]
Well, of course from our lab it would be great to have support for dotnet soon, and we will really need it for one publication on early next year.
On the other hand, I imagine that we may have been the first ones asking for support for dotnet conda package (as it has been really recently published) but we will not be the last ones given the importance of that platform...

However, if it is not possible right now to give support for it, I will temporarily remove the dotnet dependency (this implies to disable MetaMorpheus support) in order to reenable it again when bioconda supports dotnet.

**Comment by CarlosHorro** [comment]
So, will be finally wait until early next year for supporting last glibc version?
Just to do the removal...

**Comment by mbargull** [comment]
> However, if it is not possible right now to give support for it, I will temporarily remove the dotnet dependency (this implies to disable MetaMorpheus support) in order to reenable it again when bioconda supports dotnet.

If that's an option for you, I'd go with that to avoid delaying you further.
I do plan to update our build images in a few weeks, but I can't give an exact ETA currently.

**Comment by CarlosHorro** [comment]
Weird, when running the tests, doing the macos testing it shows an "infrastructure fail" :-?

**Comment by bgruening** [comment]
Something is fishy on OSX. It takes 5h to download? Or install?

**Comment by dpryan79** [comment]
@bgruening I'm killing that since it's obviously broken for some unknown reason and probably holding up unknown PRs.

**Comment by CarlosHorro** [comment]
@bgruening just to mention that I've been able to execute CircleCI in my computer, although the last outputs seemed weird:
```
...
14:17:48 BIOCONDA INFO Command closed STDOUT/STDERR but is still running
14:17:48 BIOCONDA INFO Waiting 30 seconds (1/5)

14:17:48 BIOCONDA INFO TEST SUCCESS recipes/peptide-shaker
14:17:48 BIOCONDA INFO (COMMAND) conda build purge
14:17:49 BIOCONDA INFO Command closed STDOUT/STDERR but is still running
14:17:49 BIOCONDA INFO Waiting 30 seconds (1/5)
14:17:49 BIOCONDA INFO BUILD SUMMARY: successfully built 2 of 2 recipes

```

I think I will try to run again the tests increasing the version numbers, previous problem may have been a circumstantial error.

**Comment by CarlosHorro** [comment]
@bgruening Slowly but surely ;-)

**Review by bgruening — APPROVED** [review]

**Comment by bgruening** [comment]
@bioconda-bot please merge

**Comment by bgruening** [comment]
Mühsam ernährt sich das Eichhörnchen :)

**Pull request merged**